### PR TITLE
Katana for macOS

### PR DESCRIPTION
**Pull request by smaramba**

### CODE REVIEW

File: Katana.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -18,8 +18,54 @@
 		654994B11DD31C840057D4BB /* KatanaElements.framework in Embed Frameworks */ = {isa = PBXBuildFile; fileRef = 6513094F1DC3529F00007B6B /* KatanaElements.framework */; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
 		654994C31DD4D4F20057D4BB /* CounterScreen.swift in Sources */ = {isa = PBXBuildFile; fileRef = 654994C21DD4D4F20057D4BB /* CounterScreen.swift */; };
 		6562656A1DD1C175005E91DC /* NativeButton.swift in Sources */ = {isa = PBXBuildFile; fileRef = 656265691DD1C175005E91DC /* NativeButton.swift */; };
-		65CAC5AB1DD622CC00395FE1 /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 65CAC5AA1DD622CC00395FE1 /* LaunchScreen.storyboard */; };
-		8133155B1D7B726000FBE210 /* DrawableContainer.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8133151F1D7B726000FBE210 /* DrawableContainer.swift */; };
+		65A7223B1DDF5BAF00013E7A /* PlatformNativeView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 65A7223A1DDF5BAF00013E7A /* PlatformNativeView.swift */; };
```
Comment (lucaquerella) [inline Katana.xcodeproj/project.pbxproj]: I provided you some feedback on file file structure online that I couldn't give you here.

File: Katana iOS/FloatEdgeInsets+UIEdgeInsets.swift [hunk 2]
```diff
@@ -0,0 +1,14 @@
+//
+//  FloatEdgeInsets+UIEdgeInsets.swift
+//  Katana
+//
+//  Copyright © 2016 Bending Spoons.
+//  Distributed under the MIT License.
+//  See the LICENSE file for more information.
+
+
+public extension FloatEdgeInsets {
```
Comment (lucaquerella) [inline Katana iOS/FloatEdgeInsets+UIEdgeInsets.swift]: AS discuss offline I'd go for a typealias

File: Katana macOS/NSView+Utils.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    set {
+      self.wantsLayer = true
+      self.layer?.backgroundColor = newValue?.cgColor
+    }
+  }
+}
+
+// MARK: NSView + customTag
+import ObjectiveC
+var key: ()?
```
Comment (lucaquerella) [inline Katana macOS/NSView+Utils.swift]: I'd put it inside the extension of NSView

**Comment by smaramba** [comment]
changes are up

**Comment by lucaquerella** [comment]
Great, there are tho few little things that I'd fine tuning. But before that I think we should make a decision: how many framework do we want to create?

I guess we can go for:
- KatanaIOS
- KatanaMacOS
- KatanaIOSElements
- KatanaMacOSElements

or for 

I guess we can go for:
- Katana (both iOS and macOS)
- KatanaElements (both iOS and macOS)

for a mix
- Katana (both iOS and macOS)
- KatanaIOSElements
- KatanaMacOSElements

when skyping we were leaning towards the second options but I still see both targets. is that WIP or you changed your mind? what are pros and cons? TBH I'm not 100% convinced on what's the best solution.

**Comment by bolismauro** [comment]
No strong opinion for what concern the framework composition.
Maybe having Katana that is able to work both on iOS and Mac would be better.
The only doubt I have is about testing (because I'm ignorant probably). Can we test katana both on iOS and Mac?

**Comment by smaramba** [comment]
I agree having one single Katana framework for iOS and macOS is better. For what concern the tests we can have them but we need to: a. restructure the tests to not include UIView like now (maybe using a fake View) or b. add tests for the macOS part

**Comment by lucaquerella** [comment]
I'll try to list hear what I feel it's missing so we have a TODO list before we merge in master. Feel free to challenge it.

- [x] Update the readme
- [ ] Update the pod spec
- [ ] Test the integration with Carthage
- [ ] Write macOS test target
- [ ] Write macOS elements target (with View and Button) 
- [ ] Make sure CI test both platform
- [ ] Create a simple Demo (or update HN to support both platform)

**Comment by smaramba** [comment]
added test for macOS (tests for deallocation are not passing @lucaquerella, please have a look)

**Comment by smaramba** [comment]
- added View and Button Elements for macOS
- updated the Katana macOS Coordinate System to match iOS
- updated the demo for macOS

**Pull request merged**